openthread net_mgmt_event_callback expects event info.

**Describe the bug**

When a feature enables `CONFIG_NET_MGMT` (for example `CONFIG_NET_DHCPV4`) openthread will fail to compile with the error:

```
Building C object zephyr/subsys/net/l2/openthread/CMakeFiles/subsys__net__ip__l2__openthread.dir/openthread.c.obj
...zephyr/subsys/net/l2/openthread/openthread.c: In function 'ipv6_addr_event_handler':
...zephyr/subsys/net/l2/openthread/openthread.c:125:8: error: 'struct net_mgmt_event_callback' has no member named 'info'
  125 |  if (cb->info == NULL || cb->info_length != sizeof(struct in6_addr)) {
      |
```

This is happening with version 3.0.99, but appears to be unchanged.

The [code in openthread.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/l2/openthread/openthread.c#L113=) is wrapped with a check for `CONFIG_NET_MGMT_EVENT`, but then goes on to access fields that are only available if `CONFIG_NET_MGMT_EVENT_INFO` is also set.

As a workaround `CONFIG_NET_MGMT_EVENT_INFO=y` can be specified directly in the project config.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: nrf 2.0.0 using gcc-arm-embedded to compile

--- a/subsys/net/l2/openthread/openthread.c
+++ b/subsys/net/l2/openthread/openthread.c
@@ -116,11 +116,12 @@ static struct net_mgmt_event_callback ip6_addr_cb;
 static void ipv6_addr_event_handler(struct net_mgmt_event_callback *cb,
 				    uint32_t mgmt_event, struct net_if *iface)
 {
-	struct openthread_context *ot_context = net_if_l2_data(iface);
-
 	if (net_if_l2(iface) != &NET_L2_GET_NAME(OPENTHREAD)) {
 		return;
 	}
+
+#ifdef CONFIG_NET_MGMT_EVENT_INFO
+	struct openthread_context *ot_context = net_if_l2_data(iface);
 
 	if (cb->info == NULL || cb->info_length != sizeof(struct in6_addr)) {
 		return;
@@ -131,8 +132,12 @@ static void ipv6_addr_event_handler(struct net_mgmt_event_callback *cb,
 	} else if (mgmt_event == NET_EVENT_IPV6_MADDR_ADD) {
 		add_ipv6_maddr_to_ot(ot_context, (const struct in6_addr *)cb->info);
 	}
+#else
+	NET_WARN("No address info provided with event, "
+		 "please enable CONFIG_NET_MGMT_EVENT_INFO");
+#endif /* CONFIG_NET_MGMT_EVENT_INFO */
 }
-#endif
+#endif /* CONFIG_NET_MGMT_EVENT */
 
 static int ncp_hdlc_send(const uint8_t *buf, uint16_t len)
 {

